File: 20. Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) 
    {
        int flag = 0;
        vector<char> vec;
        
        if(s.size()==0)
            return false;
        
        for(auto i = s.begin(); i != s.end() ; i++)
        {
            if(*i == '(')
            {
                vec.push_back(')');
            }
            else if(*i == '[')
            {
                vec.push_back(']');
            }
            else if(*i == '{')
            {
                vec.push_back('}');
            }
            else
            {
                if(vec.size()==0)
                    return false;
                
                if(vec[vec.size() - 1] != *i)
                {
                    return false;
                    break;
                }
                else
                {
                    vec.pop_back();
                }
                
            }
        }
        
        if(vec.size() != 0)
            return false;
        else
            return true;
        
    }
    
    
};

File: 20. Valid Parentheses.java
class Solution {
    public boolean isValid(String s) 
    {
        Stack<Character> sta = new Stack<Character>();
            
        
        
        if(s.length() == 0)
            return false;
        
        for(int i = 0; i < s.length(); i++)
        {
            
            if(s.charAt(i) == '(')
            {
                sta.push(')');
            }
            else if(s.charAt(i) == '[')
            {
                sta.push(']');
            }
            else if(s.charAt(i) == '{')
            {
                sta.push('}');
            }
            else
            {
                if(sta.empty())
                    return false;
                
                if(sta.peek().charValue() != s.charAt(i))
                {
                    return false;
                }
                else
                {
                    sta.pop();
                }
                
            }
        }
        
        if(sta.empty())
            return true;
        else
            return false;
    }
}